spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -111,23 +122,60 @@ public JSONObject getQueryResponse(AsyncQueryJobMetadata asyncQueryJobMetadata)
       String error = items.optString(ERROR_FIELD, "");
       result.put(ERROR_FIELD, error);
     } else {
-      // make call to EMR Serverless when related result index documents are not available
-      GetJobRunResult getJobRunResult =
-          emrServerlessClient.getJobRunResult(
-              asyncQueryJobMetadata.getApplicationId(), asyncQueryJobMetadata.getJobId());
-      String jobState = getJobRunResult.getJobRun().getState();
-      result.put(STATUS_FIELD, jobState);
-      result.put(ERROR_FIELD, "");
+      if (asyncQueryJobMetadata.getSessionId() != null) {
```

Comment:
so this means the feature must be enabled at the moment?

Reply:
yes. later, we can decide whether we want to enable by default.